navapbc/template-infra
# feat: add alerts and sns topic

## Ticket

https://github.com/navapbc/template-infra/issues/328

## Changes
Added monitoring module with list of most common alerts and sns topic for notifications.
Added module call from /app/service/ with default alert thresholds values
Every alert could be enabled/disabled by setting/unsetting alert threshold.

## Context for reviewers
Based on the monitoring project requirements, I incorporated a monitoring module into our existing Terraform code. The goal is to enhance our infrastructure by adding basic ECS metrics alerts with configurable thresholds. 

## Testing

### Test 1 
- Run command to deploy app and monitoring. 
`TF_CLI_ARGS_apply="-var=image_tag=8a1ebfde68391049d022f6c959557cafb179869e" make infra-update-app-service APP_NAME=app ENVIRONMENT=staging`
Ensure all alerts have been created via AWS UI - passed
### Test 2
- Set alert threshold value that will trigger alert. 
- Manually add your email as subscriber to SNS topic created by this PR. 
- Approve notification by pressing the link sent to email from AWS SNS service.
- Run command to apply terraform changes
Ensure alert is triggered via AWS UI. Ensure email is received notifying about triggered alert.  - passed
`TF_CLI_ARGS_apply="-var=image_tag=8a1ebfde68391049d022f6c959557cafb179869e" make infra-update-app-service APP_NAME=app ENVIRONMENT=staging`
### Test 3
- Uncomment threshold for any alert in /app/service/main.tf -> module "monitoring". 
- Run command to apply terraform changes 

`TF_CLI_ARGS_apply="-var=image_tag=8a1ebfde68391049d022f6c959557cafb179869e" make infra-update-app-service APP_NAME=app ENVIRONMENT=staging`
Ensure corresponding alert is created.  - passed

TargetResponseTime:
<img width="1300" alt="image" src="https://github.com/navapbc/template-infra/assets/5009912/d8b5dea1-5615-4d3c-9817-6acf004bf5bb">

app-staging-high-http-target-5xx-error-count-threshold:
<img width="1308" alt="image" src="https://github.com/navapbc/template-infra/assets/5009912/00044be9-3c81-48ba-8a45-e03ae51f740e">

app-staging-high-http-elb-5xx-error-count-threshold:
<img width="1305" alt="image" src="https://github.com/navapbc/template-infra/assets/5009912/93274608-1fa7-44b5-a9d7-a090b4ae7e38">
